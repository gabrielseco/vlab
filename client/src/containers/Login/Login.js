import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Login.scss';

import { AuthService } from 'services';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      login: '',
      password: ''
    };
    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.authService = new AuthService();
  }

  async onSubmit(evt) {
    evt.preventDefault();
    // TODO TRY CATCH
    const { token } = await this.authService.signIn(this.state)

    this.authService.setToken(token);

    /* this.router.navigateTo('/home'); */
  }

  onChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value
    });
  }

  render() {
    return (
      <div className={styles.loginContainer}>
        <div className={styles.loginHeader}>
          <h1>Login</h1>
        </div>
        <form className={styles.loginForm} onSubmit={this.onSubmit}>
          <label htmlFor="login">
            Usuario
          </label>
          <input type="text" name="login" value={this.state.login} onChange={this.onChange} />
          <label htmlFor="password">
            Contrase√±a
          </label>
          <input type="password" name="password" value={this.state.password} onChange={this.onChange}/>
          <button type="submit">Log In</button>
        </form>
      </div>
    );
  }
}

Login.propTypes = {};

export default Login;
