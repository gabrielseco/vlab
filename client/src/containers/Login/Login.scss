$container-background: #eeeeee;

$subcontainer-width: 350px;
$subcontainer-background: #ffffff;

$form-control-padding: 10px;
$form-control-margin-bottom: 10px;

.container {
  align-items: center;
  background: $container-background;
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  text-align: center;
}

.subContainer {
  background: $subcontainer-background;
  display: flex;
  flex-direction: column;
  padding: 20px;
  width: $subcontainer-width;
}

.loginForm {
  display: flex;
  flex-direction: column;

  input {
    margin-bottom: $form-control-margin-bottom;
    padding: $form-control-padding;
  }

  button {
    padding: $form-control-padding;
  }
}
