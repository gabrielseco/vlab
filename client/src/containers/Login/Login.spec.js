import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Login } from './Login';

describe('Login suite', () => {
  const component = shallow(<Login />);
  it('renders Login without any state injected', () => {
    expect(component).toBeDefined();
  });

  it('should call the onChange event', () => {
    const value = 'gabriel';
    const usernameField = component.find('.input__username_js');
    usernameField.simulate('change', {
      target: {
        name: 'username',
        value: value
      }
    });

    expect(component.state().username).toBe(value);
  });

  it('should create the snapshot of the login', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
