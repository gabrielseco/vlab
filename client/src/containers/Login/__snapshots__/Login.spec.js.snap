// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login suite should create the snapshot of the login 1`] = `
<div
  className="container"
>
  <div
    className="subContainer"
  >
    <div
      className="loginHeader"
    >
      <h1>
        Login
      </h1>
    </div>
    <form
      className="loginForm"
      onSubmit={[Function]}
    >
      <input
        className="input__username_js"
        name="username"
        onChange={[Function]}
        placeholder="Usuario"
        type="text"
        value="gabriel"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="ContraseÃ±a"
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Log In
      </button>
    </form>
  </div>
</div>
`;
